src/app/settings/keyboard-mapping-settings/components/edit-keyboard-mapping/edit-keyboard-mapping.component.html

```diff
@@ -1,12 +1,47 @@
 <sofie-row class="mb-3">
   <form [formGroup]="actionTriggerForm" class="left-container">
     <div [formGroupName]="'data'">
-      <sofie-text-input i18n-label label="global.label" [formControlName]="'label'"></sofie-text-input>
-      <sofie-keyboard-input [isRequired]="true" i18n-label label="action-triggers.shortcut.label" [formControlName]="'keys'" i18n-errorMessage errorMessage="action-triggers.shortcut.error">
+      <sofie-text-input
+        i18n-label
+        label="global.label"
+        [formControlName]="'label'">
+      </sofie-text-input>
+      <sofie-keyboard-input
+        [isRequired]="true"
+        i18n-label
+        label="action-triggers.shortcut.label"
+        [formControlName]="'keys'"
+        i18n-errorMessage
+        errorMessage="action-triggers.shortcut.error">
       </sofie-keyboard-input>
-      <sofie-keyboard-input i18n-label label="action-triggers.physical-mapping.label" [formControlName]="'mappedToKeys'" i18n-helpText helpText="action-triggers.physical-mapping-help.tooltip">
+      <sofie-button-group>
+        <sofie-button #toggleButton (click)="toggleKeyboardVisibility()" i18n>
+          edit-keyboard-mapping.toggle-virtual-keyboard.button
+        </sofie-button>
+        <sofie-spacer></sofie-spacer>
+      </sofie-button-group>
+      <div class="wrapper">
+        <div *ngIf="isKeyboardVisible" class="floating-keyboard">
+          <sofie-virtual-keyboard class="sofie-virtual-keyboard" (keyPressed)="buildKeyCombo($event)" [keyBindings]="keyBindings" [keystrokes]="keystrokes"></sofie-virtual-keyboard>
+        </div>
+      </div>
```

Comment:
The UI could be looking a bit nicer:
![image](https://github.com/tv2/sofie-web-client/assets/23049647/a82066bd-86cf-434b-97d1-7e92801dcf91)
There is now this large "gap" between your toggle button and the next input, which doesn't really make sense. It would be nice to if we could have the same distance in the gap between inputs. It gives it a more clean and "neat-looking" feeling.

Your toggle button also looks out of place.
It would be nice if it was icon (maybe the questionmark icon the others have - or better yet, a keyboard icon) that was placed above the input to the right (like with "Override key color").
When hovered, you should display "Toggle virtual keyboard" in a tooltip and when clicked, you should open the keyboard. Remember, this feature is to be used in an edgecase, so it shouldn't be taking up a significant portion of the UI.

Also, it would be nice that when you click outside of the virtual keyboard that it will close itself again. That' going to be even more necessary when the toggle isn't this big button, but a small icon instead.

Reply:
The way I see it I have two choices here:
1) Put a font-awesome faKeyboard icon behind the label
2) Put a font-awesome faKeyboard icon at the tail end of the input field

If I opt for route 2, you probably want me to create a specialized component that has all that jazz baked into it, amiright, but option one is just adding a font awesome icon behind the label?

Reply:
I found Sofie-icon-button and will use that

Reply:
Okay @LindvedKrvang. I have fixed and pushed this (but not everything you commented on). At the moment, since this functionality is a one-off occurrence, I used CSS to position it instead of tweaking the sofie-keyboard-input.component.ts file. Reason: One-off fix and not something that will be used elsewhere.

Reply:
Ref: Close when clicking outside: 
Currently It is nice to be able to click in the shortcuts textarea and reset it, use a combination if keyboard and virtual keyboard, etc. If we implement a click out that will close the dialog if we attempt to "X out" the current shortcut, etc.